includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -594,6 +596,62 @@ public function display_test_mode_notice() {
 		}
 	}
 
+	/**
+	 * Add notices explaining how to enable Apple Pay.
+	 *
+	 * @return void
+	 */
+	public function display_not_supported_apple_pay() {
+		if ( 'yes' !== $this->get_option( 'payment_request' ) ) {
+			return;
+		}
+
+		$is_in_supported_countries               = in_array( $this->account->get_account_country(), WC_Payments_Utils::supported_applepay_country_codes(), true );
+		$well_known_dir                          = untrailingslashit( ABSPATH ) . '/' . WC_Payments_Apple_Pay_Registration::DOMAIN_ASSOCIATION_FILE_DIR;
+		$fullpath                                = $well_known_dir . '/' . WC_Payments_Apple_Pay_Registration::DOMAIN_ASSOCIATION_FILE_NAME;
+		$has_domain_association_file_permissions = is_dir( $well_known_dir ) &&
+			( substr( sprintf( '%o', fileperms( $well_known_dir ) ), -4 ) === '0755' ) &&
+			file_exists( $fullpath ) &&
+			substr( sprintf( '%o', fileperms( $fullpath ) ), -4 ) === '0644';
+
+		if ( $is_in_supported_countries && $has_domain_association_file_permissions ) {
+			return;
+		}
+		if ( ! $is_in_supported_countries ) {
+			?>
+			<div id="wcpay-applepay-error" class="notice notice-error">
+				<p>
+					<b><?php esc_html_e( 'Apple Pay: ', 'woocommerce-payments' ); ?></b>
+					<?php esc_html_e( 'You are not in the countries and regions that support Apple Pay. You can find a full list of countries and regions at', 'woocommerce-payments' ); ?>
```

Comment:
Would the following be a good alternative `Your account is registered in a country or region not supported by Apply Pay.` where `not supported` is the link.